neigh_linux.go

```diff
@@ -174,40 +175,47 @@ func neighHandle(neigh *Neigh, req *nl.NetlinkRequest) error {
 	return err
 }
 
-// NeighList gets a list of IP-MAC mappings in the system (ARP table).
+// NeighList returns a list of IP-MAC mappings in the system (ARP table).
 // Equivalent to: `ip neighbor show`.
 // The list can be filtered by link and ip family.
 func NeighList(linkIndex, family int) ([]Neigh, error) {
 	return pkgHandle.NeighList(linkIndex, family)
 }
 
-// NeighProxyList gets a list of neighbor proxies in the system.
+// NeighProxyList returns a list of neighbor proxies in the system.
 // Equivalent to: `ip neighbor show proxy`.
 // The list can be filtered by link and ip family.
 func NeighProxyList(linkIndex, family int) ([]Neigh, error) {
 	return pkgHandle.NeighProxyList(linkIndex, family)
 }
 
-// NeighList gets a list of IP-MAC mappings in the system (ARP table).
+// NeighListFiltered returns a list of neighbour entries filtered by link, ip family, flag and state.
+func NeighListFiltered(linkIndex, family, flags, state int) ([]Neigh, error) {
+	return pkgHandle.NeighListFiltered(linkIndex, family, flags, state)
+}
+
+// NeighList returns a list of IP-MAC mappings in the system (ARP table).
 // Equivalent to: `ip neighbor show`.
 // The list can be filtered by link and ip family.
 func (h *Handle) NeighList(linkIndex, family int) ([]Neigh, error) {
-	return h.neighList(linkIndex, family, 0)
+	return h.NeighListFiltered(linkIndex, family, 0, 0)
 }
 
-// NeighProxyList gets a list of neighbor proxies in the system.
+// NeighProxyList returns a list of neighbor proxies in the system.
 // Equivalent to: `ip neighbor show proxy`.
 // The list can be filtered by link, ip family.
 func (h *Handle) NeighProxyList(linkIndex, family int) ([]Neigh, error) {
-	return h.neighList(linkIndex, family, NTF_PROXY)
+	return h.NeighListFiltered(linkIndex, family, NTF_PROXY, 0)
 }
 
-func (h *Handle) neighList(linkIndex, family, flags int) ([]Neigh, error) {
+// NeighListFiltered returns a list of neighbour entries filtered by link, ip family, flag and state.
+func (h *Handle) NeighListFiltered(linkIndex, family, flags, state int) ([]Neigh, error) {
```

Comment:
should the `state` field be `uint16` or do we have a convention in this project where exported method simply take a `int` parameter and caller is supposed to know the value range ?

Reply:
I like more the uint16 convention